app/scripts/controllers/replicaSet.js

```diff
@@ -0,0 +1,448 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:ReplicaSetController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('ReplicaSetController',
+              function ($scope,
+                        $filter,
+                        $routeParams,
+                        AlertMessageService,
+                        DataService,
+                        HPAService,
+                        MetricsService,
+                        ProjectsService,
+                        DeploymentsService,
+                        ImageStreamResolver,
+                        Navigate,
+                        keyValueEditorUtils,
+                        kind) {
+    var hasDC = false;
+    var displayKind = $filter('humanizeKind')(kind);
+    switch (kind) {
+    case 'ReplicaSet':
+      $scope.breadcrumbs = [
+        {
+          title: "Replica Sets",
+          link: "project/" + $routeParams.project + "/browse/deployments"
+        }
+      ];
+
+      $scope.resource = {
+        group: "extensions",
+        resource: "replicasets"
+      };
+      $scope.healthCheckURL = Navigate.healthCheckURL($routeParams.project,
+                                                      "ReplicaSet",
```

Comment:
if you passed the kind var in here instead wouldn't these two be identical and you could move this out of the switch statement?